### PR TITLE
WIP: move redux state manipulation to the editor

### PR DESCRIPTION
**Pull request by bcolloran**
this is a very WIP PR to simplify our codebase by moving all of the redux state manipulation to the editor.

as of the initial PR and this comment, i have:
* expanded the editor state schema to include everything that used to be in the eval frame
* removed the eval frame only state schema stuff
* added a eval frame state selector function that plucks out the bits of state that need to be forwarded to the eval frame
* basically eliminated the reducers in the eval frame

for the time being, this kind of inverts the previous paradigm; all the redux actions in the eval frame side are still dispatched to redux, but the eval frame does not have any reducer cases to handle them within the eval fram; instead, the actions are passed to the editor's redux store, which now has reducer cases to update the store accordingly, and after that, the evalFrameStateSelector function is called on the current state, and all of the relevant state is sent en bloc to the eval frame, where the state is replaced entirely.

the work that is left to be done is to unentangle the redux store in the eval frame from the passing of actions -- i think we should not need to use "dispatch" in the eval frame any more; instead, we can pass messages directly back up to the editor. the other part of the work will be sanitizing/validating the passed messages (for security)

## remaining to-dos before merging:
- [ ] implement white-listing of actions
- [x] fix console input behavior
- [x] clear warning/logs

### CODE REVIEW

**Comment by hamilton** [comment]
High-level, this feels like the right approach, and will hopefully finally cut out the last bit of overhead and complexity from the iframe migration.

In the airport so will probably do another pass over this codebase once I'm back home.

It'd be great to have more of an enumeration of what needs to be done (even a list with the `- [ ]` syntax in the original comment is sufficient and keeps everything in one place).

One other thing to check - this breaks plugin loads w/ "run all" (rather, cells that use the language plugin error out now on "run all", whereas `master` they don't). There are probably more bugs that'll need to be more thoroughly checked, but that was the first that I noticed.

Let's try to make this the last "hold off doing any real work until this one lands" style of PR going forward. I don't recall the team discussing if this was in-scope right now, so this rubs me a bit the wrong way, especially with more critical bugs & rough edges relating to the user experience that will surely be uncovered once we roll out alpha. I'm all for the change, just not for the approach. That all said, the train has left the station, so let me know how I can help get this PR landed via other PRs targeting it.

File: src/state-schemas/eval-frame-state-selector.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import _ from 'lodash'
```
[the remark this answers is not in the document]
Comment (bcolloran) [reply src/state-schemas/eval-frame-state-selector.js]: oh good point, added #1265

File: src/state-schemas/state-prototype-from-schema.js [hunk 2]
```diff
@@ -1,18 +1,5 @@
 import _ from 'lodash'
```
[the remark this answers is not in the document]
Comment (bcolloran) [reply src/state-schemas/state-prototype-from-schema.js]: as above, see #1265

**Comment by bcolloran** [comment]
remaining to-dos before merging:
- [ ] fix console input behavior
- [ ] implement white-listing of actions

**Pull request merged**